app/grandchallenge/evaluation/models.py

```diff
@@ -1854,6 +1869,58 @@ def inputs_complete(self):
         else:
             return False
 
+    @property
+    def additional_inputs(self):
+        # additional inputs as currently defined on the phase
+        phase_input_slugs = self.submission.phase.inputs.values_list(
+            "slug", flat=True
+        )
+        return self.inputs.filter(interface__slug__in=phase_input_slugs)
```

Comment:
I'm using this to display the inputs on the detail page. The way I implemented it now, if half way through the challenge, the inputs are removed from the Phase, they would not show for any of the evaluations, even if those evaluations do have these inputs. 

I could alternatively, collect a list of other input slugs (predictions-json, predictions-csv, algorithm-outputs defined on the phase) and exclude those from the inputs rather than subsetting by inputs currently set on the phase. Then the inputs would still show for old evaluations even if the phase configuration changes.

Reply:
I think given the current state of this part of the codebase this is fine. Maybe the field `phase.inputs` should not be used at all and we should drop it and start with a new thing, `phase.evaluation_inputs`, that handles things in the standard way, I don't know.

Reply:
It is indeed a bit confusing, but I think it will always be because the evaluation inputs will always be a combination of standard handled inputs (validated and added on submission creation from user input) and a set of algorithm outputs, which are added at a later point in time.

Reply:
For now, I will leave it at this.